extjsdk/src/main/java/io/vantiq/extjsdk/InstanceConfigUtils.java

```diff
@@ -0,0 +1,187 @@
+import static io.vantiq.extjsdk.Utils.SERVER_CONFIG_DIR;
+import static io.vantiq.extjsdk.Utils.SERVER_CONFIG_FILENAME;
+import static io.vantiq.extjsdk.Utils.SOURCES_PROPERTY_NAME;
+import static io.vantiq.extjsdk.Utils.TARGET_SERVER_PROPERTY_NAME;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Properties;
+
+public class InstanceConfigUtils {
+
+    // String used by methods to synch on
+    private final String SYNCH_LOCK = "synchLockString";
+    
+    public static final String CONFIG_WAS_PROGRAMMATIC = "configWasProgrammatic";
+
+    // The properties object containing the data from the server configuration file
+    private Properties serverConfigProperties;
+    
+    /**
+     * Provide server config properties programmatically.
+     *
+     * Used when the configuration properties necessary to connect to Vantiq are provided via a means outside
+     * the reading of the server.config file.  This also sets a property telling this class that the information
+     * was provided programmatically so that subsequent calls to obtainServerConfig will not overwrite things from
+     * the server.config file or generate an error when that file doesn't exist.
+     *
+     * If this is called repeatedly, each will override the last.
+     *
+     * @param targetServer  String URL for the Vantiq server
+     * @param authToken     String indicating the access token to use to connect to the server at targetServer
+*    * @param sources       String containing a comma separated list of source names to use.
+     * @param sendPings     Boolean indicating whether the client connection should ping the Vantiq server to maintain
+     *                      the connection.
+     * @param tcpProbePort  Integer providing a port to open for TCP probes when running in a Kubernetes environment.
+     *
+     * @return              Properties the server config properties as provided.
+     */
+    public Properties provideServerConfig(String targetServer, String authToken, String sources,
+                                                 Boolean sendPings, Integer tcpProbePort) {
+        Properties localSCP = new Properties();
+        if (StringUtils.isAnyEmpty(targetServer, authToken, sources)) {
+            throw new IllegalArgumentException("The targetServer, authToken, and sources parameters " +
+                    "must be non-null & non-empty.");
+        }
+        localSCP.put(TARGET_SERVER_PROPERTY_NAME, targetServer);
+        localSCP.put(AUTH_TOKEN_PROPERTY_NAME, authToken);
+        localSCP.put(SOURCES_PROPERTY_NAME, sources);
+        localSCP.put(SEND_PING_PROPERTY_NAME, (sendPings == null ? Boolean.FALSE.toString() : sendPings));
+        if (tcpProbePort != null) {
+            localSCP.put(PORT_PROPERTY_NAME, tcpProbePort.toString());
+        }
+        localSCP.put(CONFIG_WAS_PROGRAMMATIC, true);
```

Comment:
Mark that this instance was programmatically configured so that repeated calls to `obtainServerConfig()` to not attempt to re-read a non-existent file.